testsuite/tests/typing-layouts-arrays/generate_makearray_dynamic_tests.ml

```diff
@@ -0,0 +1,678 @@
+let ty_ur3 = Ty.(unboxed_record "ur3" ["a", int64_u])
+let ty_ur4 = Ty.(unboxed_record "ur4" ["a", ty_ur1; "b", ty_ur3])
+
+(* Types the GC always ignores, which can be used with %makearray_dynamic_uninit *)
+let always_ignored_types = Ty.([
+  float32_u; float_u; int32_u; int64_u; nativeint_u; ty_ur1; ty_ur3; ty_ur4;
+  unboxed_tuple [float_u; int32_u; int64_u];
+  unboxed_tuple [
+    float_u;
+    unboxed_tuple [int64_u; int64_u];
+    float32_u;
+    unboxed_tuple [int32_u; unboxed_tuple [float32_u; float_u]];
+    int64_u;
+  ];
+  unboxed_tuple [int64_u; ty_ur1];
+])
+
+let types = always_ignored_types @ Ty.([
+  float32; float; int32; int64; nativeint; int; enum 3; ty_ur2;
+  unboxed_tuple [int; int64];
+  unboxed_tuple [
+    option int64;
+    unboxed_tuple [int; int32; float];
+    float;
+    unboxed_tuple [float32; option (tuple [nativeint; nativeint])];
+    int32
+  ];
+  unboxed_tuple [float; float; float];
+  unboxed_tuple [
+    float;
+    unboxed_tuple [float; float];
+    unboxed_tuple [float; unboxed_tuple [float; float; float]]
+  ];
+  unboxed_tuple [float_u; int; int64_u];
+  unboxed_tuple [
+    float_u;
+    unboxed_tuple [int; int64_u];
+    float32_u;
+    unboxed_tuple [int32_u; unboxed_tuple [float32_u; float_u]];
+    int;
+  ];
+  unboxed_tuple [ty_ur2; ty_ur1];
+])
+
+let preamble = {|
+open Stdlib_upstream_compatible
+open Stdlib_stable
+module List = ListLabels
+module String = StringLabels
+
+external[@layout_poly] makearray_dynamic_uninit_local :
+  ('a : any_non_null) . int -> 'a array @ local =
+  "%makearray_dynamic_uninit"
+
+external[@layout_poly] makearray_dynamic_uninit :
+  ('a : any_non_null) . int -> 'a array =
+  "%makearray_dynamic_uninit"
+
+external[@layout_poly] makearray_dynamic_local :
+  ('a : any_non_null) . int -> 'a -> 'a array @ local =
+  "%makearray_dynamic"
+
+external[@layout_poly] makearray_dynamic :
+  ('a : any_non_null) . int -> 'a -> 'a array =
+  "%makearray_dynamic"
+
+external[@layout_poly] get :
+  ('a : any_non_null) . ('a array[@local_opt]) -> (int[@local_opt]) -> 'a =
+  "%array_safe_get"
+
+external[@layout_poly] set :
+  ('a : any_non_null) . ('a array[@local_opt]) -> (int[@local_opt]) -> 'a -> unit =
+  "%array_safe_set"
+
+external[@layout_poly] unsafe_blit :
+  ('a : any_non_null) . ('a array[@local_opt]) -> (int[@local_opt]) -> ('a array[@local_opt]) -> (int[@local_opt]) -> (int[@local_opt]) -> unit =
+  "%arrayblit"
+
+let failwithf fmt = Printf.ksprintf failwith fmt
+
+external globalize : local_ 'a -> 'a = "%obj_dup";;
+
+(* Redefine iter to infer locality *)
+let rec iter ~f = function
+    [] -> ()
+  | a::l -> f a; iter ~f l
+
+let tests_run = ref []
+
+let mark_test_run test_id =
+  if not (List.mem ~set:!tests_run test_id) then
+    tests_run := test_id :: !tests_run
+
+(* Various interesting values *)
+
+let sizes = [ 0; 1; 2; 30; 31; 32 ]
+
+let bad_indices size = [ -100; -2; -1; size; size + 1; size + 100 ]
```

Comment:
I think this list should include `min_int` and `max_int`.

Reply:
Thanks, added a commit to [`rtjoa.makearray_dynamic_tests`](https://github.com/ocaml-flambda/flambda-backend/tree/rtjoa.makearray_dynamic_tests).